Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6738.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Xamarin.Forms.CustomAttributes;
+using Xamarin.Forms.Internals;
+using Xamarin.Forms.Controls;
+using System;
+
+#if UITEST
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest;
+using NUnit.Framework;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+#if UITEST
+	[Category(UITestCategories.Shell)]
+#endif
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 6738, "Flyout Navigation fails when coupled with tabs that have a stack", PlatformAffected.Android)]
+	public class Issue6738 : TestShell
+	{
+		Tab flyoutContent = new Tab();
+		Button pushPageButton = new Button { Text = "Tap to push new page to stack" };
+		Button insertPageButton = new Button { Text = "Push another page to the stack, then go to tab two" };
+		ContentPage pushedPage;
+		Tab tabOne = new Tab { Title = "TabOne" };
+		Tab tabTwo = new Tab { Title = "TabTwo " };
+
+
+		void OnReturnTapped(object sender, EventArgs e)
+		{
+			ForceTabSwitch();
+		}
+
+		void OnPushTapped(object sender, EventArgs e)
+		{
+			pushedPage = new ContentPage { Content = insertPageButton };
+			Navigation.PushAsync(pushedPage);
+		}
+
+		void OnInsertTapped(object sender, EventArgs e)
+		{
+			Navigation.InsertPageBefore(new ContentPage { Content = new Label { Text = "This is an extra page" } }, pushedPage);
+			ForceTabSwitch();
+		}
+
+		void ForceTabSwitch()
+		{
+			if (CurrentItem != null)
+			{
+				if (CurrentItem.CurrentItem == tabOne)
+				{
+					CurrentItem.CurrentItem = tabTwo;
+				}
+				else
+					CurrentItem.CurrentItem = tabOne;
+			}
+		}
+
+		protected override void Init()
+		{
+			var tabOnePage = new ContentPage { Content = pushPageButton };
+			var stackLayout = new StackLayout();
+			stackLayout.Children.Add(new Label { Text = "If you've been here already, go to tab one now. Otherwise, go to Other Page in the flyout." });
+			var returnButton = new Button { Text = "Go back to tab 1" };
+			returnButton.AutomationId = "GoToFirstTabButton";
+			returnButton.Clicked += OnReturnTapped;
+			stackLayout.Children.Add(returnButton);
+
+			var tabTwoPage = new ContentPage { Content =  stackLayout };
+			tabOne.Items.Add(tabOnePage);
+			tabTwo.Items.Add(tabTwoPage);
+
+			pushPageButton.AutomationId = "PushPageButton";
+			pushPageButton.Clicked += OnPushTapped;
+			insertPageButton.AutomationId = "InsertPageButton";
+			insertPageButton.Clicked += OnInsertTapped;
+			flyoutContent.Items.Add(new ContentPage { Content = new Label { Text = "Go back to main page via the flyout" } });
+
+			Items.Add(
+					new FlyoutItem
+					{
+						Title = "Main",
+						Items = { tabOne, tabTwo }
+					}
+			);
+			Items.Add(new FlyoutItem {
+				Title = "Other Page",
+				Items = { flyoutContent }
+			});
+		}
+
+#if UITEST
+#if !(__ANDROID__ || __IOS__)
+		[Ignore("Shell test is only supported on Android and iOS")]
+#endif
+		[Test]
+		public void Issue6738Test()
```

Comment:
- Can you rename this to indicate what's being tested? 
- This test is failing on iOS. Are you able to run UI Tests on iOS? 
- Currently the iOS tests are a bit quirky :-/ and you have to update to UI Test 3.0 in order for them to run. You don't have to update to nunit 3 just Xamarin UI test 3.0 and then they will run

If you can't get them to run locally let me know and I can see what needs to be fixed there

Here's the exception 
```
System.Exception : Error while performing Back()

  ----> System.Exception : Error while performing Query(ClassFull("_UINavigationItemButtonView"))

  ----> System.Net.WebException : POST Failed

  at Xamarin.UITest.Utils.ErrorReporting.With (System.Action func, System.Object[] args, System.String memberName) [0x00052] in <2a16c16730a54859bda72c6bc1c728f7>:0 

  at Xamarin.UITest.iOS.iOSApp.Back () [0x00000] in <2a16c16730a54859bda72c6bc1c728f7>:0 

  at Xamarin.Forms.Controls.ScreenshotConditionalApp.Back () [0x00001] in <cbd90fb7dec64bfcac03c7689d7f18c1>:0 
```

it only happens on ios11 and ios12

Reply:
This issue is only occurring in Android and the fix only applies to Android. Is it still necessary for the test to run on iOS?  I admittedly copy/pasted a lot of the boilerplate from another test and didn't adjust to only have it run on Android. I'm happy to look at fixing iOS if it's needed but I wanted to make sure we were on the same page.

Reply:
> Is it still necessary for the test to run on iOS

It is :-) And your UI Test has revealed a particularly gnarly failure on iOS :-/ 

For now I would say annotate the test so it only runs on Android,  rename the Test and I'll work on the iOS fix in a different branch

Reply:
Yeah the more I think about this the more I realize that there's no good reason not to ensure this functionality works/continues to work on iOS, thanks for talking it through. I know I have a few contributions already but it's been a while and I'm still trying to understand the project and the team's process for everything.

Reply:
Hehe, yea sometimes we get double the fun after writing the UI Tests. 

The iOS fixed is kind of an involved fix with Shell and dispose patterns  :-/  I have a basic fix but I need to vet it a bit more